querydsl-core/src/main/java/com/querydsl/core/types/dsl/PathBuilder.java

```diff
@@ -125,13 +128,15 @@ public Object getMetadata(Path<?> property) {
      */
     @SuppressWarnings("unchecked")
     public PathBuilder<Object> get(String property) {
-        PathBuilder<Object> path = (PathBuilder) properties.get(property);
+        SimpleEntry<String, Class<?>> entry = new SimpleEntry<String, Class<?>>(property, Object.class);
+        PathBuilder<Object> path = (PathBuilder<Object>) properties.get(entry);
+        PathBuilder<?> existingPath = null;
         if (path == null) {
             Class<?> vtype = validate(property, Object.class);
             path = new PathBuilder<Object>(vtype, forProperty(property), validator);
-            properties.put(property, path);
+            existingPath = properties.putIfAbsent(entry, path);
```

Comment:
Question: should we recreate the `entry` before `put()`ing, in case a validator returns a more specific `vtype`?